### PR TITLE
Xw 458

### PR DESCRIPTION
**Pull request by justnpT**
1. Merged selectors responsible for verifying if user is logged in, so that now the method is skin independent
2. Added get(url) in afterNavigate method, because the destination url was lost after logging in

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java [hunk 1]
```diff
@@ -266,7 +266,10 @@ public void afterNavigateTo(String url, WebDriver driver) {
       User user = method.getAnnotation(Execute.class).asUser();
       if (user == User.ANONYMOUS) {
```
Comment (bognix) [inline src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java]: what for is this empty if statement?

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/EditorTests.java [hunk 2]
```diff
@@ -41,13 +40,10 @@ public void beforeMethod() {
...
 
-    // This login is temporary solution, use @Execute after QAART-669 is done
-    new LoginPageObject(driver).get().logUserIn(Configuration.getCredentials().userNameStaff2,
-                                          Configuration.getCredentials().passwordStaff2);
   }
 
-  @Test(groups = "MercuryCuratedEditorTest_001", enabled = false)
-  @Execute(onWikia = MercuryWikis.MERCURY_EMPTY_CC_EDITOR)
+  @Test(groups = "MercuryCuratedEditorTest_001")
+  @Execute(onWikia = MercuryWikis.MERCURY_EMPTY_CC_EDITOR, asUser = User.STAFF)
```
Comment (bognix) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/EditorTests.java]: should those tests be enabled now?

File: src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java (hunk 1 above)
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java]: the logic results in, for Anonymous annotation do nothing, for any other case log in the user

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/EditorTests.java (hunk 2 above)
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/EditorTests.java]: yes, the test passes on production

File: src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java (hunk 1 above)
Comment (bognix) [inline src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java]: so why not having if with:
```
if (user != User.ANONYMOUS)
```
and skip else?

Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java]: +1 to @bognix

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java [hunk 3]
```diff
@@ -92,8 +90,13 @@
 public class WikiBasePageObject extends BasePageObject {
 
   protected final static By LOGIN_BUTTON_CSS = By.cssSelector("a[data-id='login']");
-  private static final String LOGGED_IN_USER_SELECTOR_VENUS =
+  private static final String LOGGED_IN_USER_SELECTOR_OASIS =
       ".AccountNavigation a[title*=%userName%]";
+  private static final String LOGGED_IN_USER_SELECTOR_MONOBOOK = "#pt-userpage a[href*=%userName%]";
+  private static final String LOGGED_IN_USER_SELECTOR_MERCURY = ".avatar img[alt*=%userName%]";
+  private static final String LOGGED_IN_USER_SELECTOR = LOGGED_IN_USER_SELECTOR_MERCURY +
+                                                        "," + LOGGED_IN_USER_SELECTOR_OASIS +
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java]: can you move comma to line above for better readability?

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java [hunk 4]
```diff
@@ -92,8 +90,13 @@
...
+                                                        "," + LOGGED_IN_USER_SELECTOR_MONOBOOK;
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java]: can you move comma to line above for better readability?

**Comment by justnpT** [comment]
Closing PR. Don't remove the Pull request and the branch

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java [hunk 5]
```diff
@@ -692,11 +676,10 @@ public void logInViaModal(String userName, String password) {
   }
 
   public String receiveMailWithNewPassword(String email, String password) {
-    String newPassword =
-        MailFunctions.getPasswordFromEmailContent((MailFunctions.getFirstEmailContent(email,
-            password, "Reset your Wikia password")));
+    String newPassword = MailFunctions.getPasswordFromEmailContent(
+        (MailFunctions.getFirstEmailContent(email, password, "Reset your Wikia password")));
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java [hunk 6]
```diff
@@ -92,8 +90,13 @@
 public class WikiBasePageObject extends BasePageObject {
 
   protected final static By LOGIN_BUTTON_CSS = By.cssSelector("a[data-id='login']");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java [hunk 7]
```diff
@@ -266,7 +266,8 @@ public void afterNavigateTo(String url, WebDriver driver) {
...
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java]: MAJOR Either remove or fill this block of code. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00108)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 34 issues:
* MAJOR 18 major
* MINOR 16 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L373) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)
* MINOR [Define a constant instead of duplicating this literal ".html'>HTML Source</a></td></tr>" 3 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L78) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L332) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L312) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
* MAJOR [Remove those useless parentheses.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L180) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)
* MINOR [Define a constant instead of duplicating this literal "</td><td>" 5 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "\"><td>" 5 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Introduce a new variable instead of reusing the parameter "imageAsBase64".](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L178) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)
* MINOR [Define a constant instead of duplicating this literal "<tr class=\"" 5 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal ".png'>Screenshot</a><br/><a href='screenshots/screenshot" 3 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L77) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "</td><td> <br/> &nbsp;</td></tr>" 6 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L140) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "error" 4 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L171) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MINOR [Define a constant instead of duplicating this literal "success" 4 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Remove this unused "date" local variable.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L381) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L414) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L415) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)
* MAJOR [Remove this unused "dateFormat" local variable.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L380) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1481)
* MINOR [Define a constant instead of duplicating this literal ".html" 3 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L72) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L949) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L740) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MINOR [Define a constant instead of duplicating this literal "%userName%" 3 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L323) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Move the array designator from the variable to the type.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L449) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1197)
* MINOR [Define a constant instead of duplicating this literal "logOut" 5 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L722) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L721) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L935) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L730) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MINOR [Define a constant instead of duplicating this literal "page loads for more than 30 seconds" 3 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L722) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal " opened" 5 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L216) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [2 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/EditorTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MINOR [Define a constant instead of duplicating this literal "Curated Content" 4 times.](https://github.com/Wikia/selenium-tests/blob/ee5538a27550352bc3cd22ea6bf7ba14fc9e1614/src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/EditorTests.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)

**Pull request closed without merging**